packages/@react-spectrum/menu/docs/Menu.mdx

```diff
@@ -0,0 +1,357 @@
+
+## Sections
+Menu supports sections in order to group options. Sections can be used by wrapping groups of Items in a `Section` component. Each `Section` takes a `title` and `key` prop.
+
+### Static Items
+
+```tsx example
+<MenuTrigger>
+  <ActionButton>
+    Edit
+  </ActionButton>
+  <Menu>
+    <Section title="Styles">
+      <Item key="bold">Bold</Item>
+      <Item key="underline">Underline</Item>
+    </Section>
+    <Section title="Align">
+      <Item key="left">Left</Item>
+      <Item key="middle">Middle</Item>
+      <Item key="right">Right</Item>
+    </Section>
+  </Menu>
+</MenuTrigger>
+```
+
+### Dynamic Items
+
+Sections used with dynamic items are populated from a hierarchical data structure. Similarly to the props on Menu, `<Section>` takes an array of data using the `items` prop.
+
+```tsx example
+function Example() {
+  let openWindows = [
+    {
+      name: 'Left Panel',
+      id: 'left',
+      children: [
+        {id: 1, name: 'Final Copy (1)'}
+      ]
+    },
+    {
+      name: 'Right Panel',
+      id: 'right',
+      children: [
+        {id: 2, name: 'index.ts'},
+        {id: 3, name: 'package.json'},
+        {id: 4, name: 'license.txt'}
+      ]
+    }
+  ];
+
+  return (
+    <MenuTrigger>
+      <ActionButton>
+        Window
+      </ActionButton>
+      <Menu
+        items={openWindows}>
+        {item => (
+          <Section items={item.children} title={item.name}>
+            {item => <Item>{item.name}</Item>}
+          </Section>
+        )}
+      </Menu>
+    </MenuTrigger>
+  );
+}
+```
+
+### Accessibility
+
+Sections without a `title` must provide an `aria-label` for accessibility.
+
+## Events
+Use the `onSelectionChange` prop as a callback to handle press events on items when selectionMode is either single or multiple. See [Selection](#selection) for more information.
+
+Menu also supports the `onAction` callback when `selectionMode` is `none` (default).
+
+```tsx example
+function Example() {
+  let [action, setAction] = React.useState('');
+
+  return (
+    <>
+      <MenuTrigger>
+        <ActionButton>
+          Edit
+        </ActionButton>
+        <Menu onAction={setAction}>
+          <Item key="cut">Cut</Item>
+          <Item key="copy">Copy</Item>
+          <Item key="paste">Paste</Item>
+        </Menu>
+      </MenuTrigger>
+      <p>Action: {action}</p>
+    </>
+  );
+}
+```
+
+## Selection
+Menu supports multiple selection modes. By default, selection is disabled, however this can be changed using the `selectionMode` prop.
+Use `defaultSelectedKeys` to provide a default set of selected items (uncontrolled) and `selectedKeys` to set the selected items (controlled). The value of the selected keys must match the `key` prop of the items.
+
+```tsx example
+<MenuTrigger>
+  <ActionButton>
+    Align
+  </ActionButton>
+  <Menu selectionMode="single" defaultSelectedKeys={['middle']}>
```

Comment:
FYI it resets when you close and open the menu again... state is lost because it unmounts the whole menu. Maybe uncontrolled isn't so useful?

Reply:
Yeah I was just looking at this one too. I'm not sure what to do with it. Would it be odd to talk about uncontrolled and controlled but only show controlled in the examples?

Reply:
mmm, no I think we might be able to get away with it...